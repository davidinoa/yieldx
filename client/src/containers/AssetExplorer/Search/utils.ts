interface InGridFilter {
  filter: string | number;
  filterTo: number;
  filterType: string;
  type: string;
  dateFrom: string;
  dateTo: string;
  values: string[];
}

export interface AgGridFilters {
  [key: string]: any;
}

const getFilter = (filter: InGridFilter) => {
  /**
   *  The filter object from Ag Grid API. In the case of filtering `name` column.
   *  { filter: "Asset Name", filterType: "text", type: "contains"}
   */
  let searchFilter = {};
  const {
    filter: filterContent,
    filterType,
    type,
    filterTo,
    values,
    dateFrom,
    dateTo
  } = filter;
  if (filterType === "text") {
    switch (type) {
      case "contains":
        searchFilter = { containsText: filterContent };
        break;
      case "notContains":
        searchFilter = { doesNotContainText: filterContent };
        break;
      case "equals":
        searchFilter = { equalsText: filterContent };
        break;
      case "notEqual":
        searchFilter = { doesNotEqualText: filterContent };
        break;
      case "startsWith":
        searchFilter = { startsWithText: filterContent };
        break;
      case "endsWith":
        searchFilter = { endsWith: filterContent };
        break;
      default:
        searchFilter = { containsText: filterContent };
        break;
    }
  } else if (filterType === "number") {
    switch (type) {
      case "equals":
        searchFilter = { equalsNumber: filterContent };
        break;
      case "notEqual":
        searchFilter = { doesNotEqualNumber: filterContent };
        break;
      case "lessThan":
        searchFilter = { isLessThanNumber: filterContent };
        break;
      case "lessThanOrEqual":
        searchFilter = { isLessThanOrEqualToNumber: filterContent };
        break;
      case "greaterThan":
        searchFilter = { isGreaterThanNumber: filterContent };
        break;
      case "greaterThanOrEqual":
        searchFilter = { isGreaterThanOrEqualToNumber: filterContent };
        break;
      case "inRange":
        searchFilter = {
          isGreaterThanOrEqualToNumber: filterContent,
          isLessThanOrEqualToNumber: filterTo
        };
        break;
      default:
        searchFilter = { equalsNumber: filterContent };
        break;
    }
  } else if (filterType === "date") {
    const convertedDateFrom = new Date(dateFrom);
    const convertedDateTo = new Date(dateTo);
    switch (type) {
      case "equals":
        searchFilter = { equalsDate: convertedDateFrom };
        break;
      case "greaterThan":
        searchFilter = { isGreaterThanDate: convertedDateFrom };
        break;
      case "lessThan":
        searchFilter = { isLessThanDate: convertedDateFrom };
        break;
      case "notEqual":
        searchFilter = { doesNotEqualDate: convertedDateFrom };
        break;
      case "inRange":
        searchFilter = {
          isGreaterThanOrEqualToDate: convertedDateFrom,
          isLessThanOrEqualToDate: convertedDateTo
        };
        break;
      default:
        searchFilter = { equalsDate: convertedDateFrom };
        break;
    }
  } else if (filterType === "set" && values) {
    // We may need to add `set` type Ag grid column filter params like cusip or isin
    // in the search request in the future.
    // searchFilter = { inText: values };
  }
  return searchFilter;
};

export const convertInGridFilters = (filters: AgGridFilters) => {
  const convertedFilters: { [index: string]: any } = {};
  Object.keys(filters).forEach(key => {
    const inGridFilter = filters[key];
    if (key === "minimumQuantity") key = "bond.minimumDenomination";
    convertedFilters[key] = getFilter(inGridFilter);
  });

  return convertedFilters;
};
