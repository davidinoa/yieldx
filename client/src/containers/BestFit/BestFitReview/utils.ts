import { getRatingValue } from "../../../util/ratings";
import YieldXPortfolio, {
  YieldXPosition
} from "../../../models/YieldXPortfolio";

export function getBondRatingsData(
  portfolio: YieldXPortfolio
): { name: string; value: number }[] {
  if (portfolio.getBondSleeve) {
    const ratingMarketValues: { [key: string]: number } = {};
    const bondPortfolio = portfolio.getBondSleeve();
    bondPortfolio.positions.forEach((position: YieldXPosition) => {
      const { asset, weight } = position;
      const rating = asset.rating ?? "NR";
      ratingMarketValues[rating] = (ratingMarketValues[rating] || 0) + weight;
    });
    return Object.entries(ratingMarketValues)
      .map(([rating, weight]) => {
        return {
          name: rating,
          value: weight
        };
      })
      .sort((a, b) => {
        return getRatingValue(a.name) - getRatingValue(b.name);
      });
  }
  return [];
}
