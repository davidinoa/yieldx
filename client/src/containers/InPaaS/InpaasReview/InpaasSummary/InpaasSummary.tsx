import React from "react";
import { Row } from "react-bootstrap";
import { Constraints } from "@bondhouse/income-portfolios";
import { ToolTipContent } from "assets/content";
import PortfolioSection from "containers/Portfolios/PortfolioDisplay/PortfolioSection/PortfolioSection";
import PortfolioMetricCard from "components/PortfolioMetricCard/PortfolioMetricCard";
import CardValue from "components/PortfolioMetricCard/CardValue/CardValue";

interface Props {
  version?: "CURRENT" | "ORIGINAL";
  reviewData?: {
    risk: number;
    maxDrawdown: number;
    actualAllocation: Required<Constraints>;
    actualYield?: number;
    actualIncome: number;
  };
  loading?: boolean;
  error?: Error;
  savedPortfolioData?: {
    expenseRatio?: number;
    risk?: number;
    maxDrawdown?: number;
    targetYield: number;
    targetIncome: number;
    actualYield?: number;
    actualIncome?: number;
  };
}

export default function InpaasSummary({
  version,
  reviewData,
  savedPortfolioData
}: Props) {
  if (reviewData)
    return (
      <PortfolioSection title="Summary" version={version}>
        <Row>
          <PortfolioMetricCard
            xs={6}
            value={{
              tooltipContent: ToolTipContent.targetYield,
              label: "Yield",
              number: reviewData.actualYield,
              format: "%",
              helperText: "annual"
            }}
          />
          <PortfolioMetricCard
            xs={6}
            value={{
              tooltipContent: ToolTipContent.targetIncome,
              label: "Income",
              number: reviewData.actualIncome,
              format: "$",
              helperText: "monthly",
              toggleLabel: "Annual"
            }}
            secondValue={{
              tooltipContent: ToolTipContent.targetIncome,
              label: "Income",
              number: reviewData.actualIncome * 12,
              format: "$",
              toggleLabel: "Monthly",
              helperText: "annual"
            }}
          />
          <PortfolioMetricCard
            xs={6}
            value={{
              tooltipContent: ToolTipContent.risk,
              label: "Risk",
              number: reviewData.risk,
              format: "%",
              toggleLabel: "Max drawdown"
            }}
            secondValue={{
              tooltipContent: ToolTipContent.maxDrawdown,
              label: "Max drawdown",
              toggleLabel: "Risk",
              number: reviewData.maxDrawdown,
              format: "%"
            }}
          />
          <PortfolioMetricCard xs={6} label="Allocation">
            <CardValue
              value={{
                format: "%",
                number: reviewData.actualAllocation.minimumWeight * 100,
                helperText: "min"
              }}
              secondValue={{
                format: "%",
                number: reviewData.actualAllocation.maximumWeight * 100,
                helperText: "max"
              }}
            />
          </PortfolioMetricCard>
        </Row>
      </PortfolioSection>
    );
  if (savedPortfolioData)
    return (
      <PortfolioSection title="Summary" version={version}>
        <Row>
          <PortfolioMetricCard
            xs={6}
            value={{
              tooltipContent: ToolTipContent.CurrentYield,
              label: "Current Yield",
              number: savedPortfolioData.actualYield,
              format: "%",
              helperText: "annual",
              toggleLabel: "%"
            }}
          />
          <PortfolioMetricCard
            xs={6}
            value={{
              tooltipContent: ToolTipContent.CurrentIncome,
              label: "Current Income",
              number: savedPortfolioData.actualIncome,
              format: "$",
              toggleLabel: "$",
              helperText: "monthly"
            }}
          />
          <PortfolioMetricCard
            xs={6}
            value={{
              tooltipContent: ToolTipContent.ExpenseRatio,
              label: "Expense ratio",
              number: savedPortfolioData.expenseRatio,
              format: "%"
            }}
          />
          <PortfolioMetricCard
            xs={6}
            value={{
              tooltipContent: ToolTipContent.risk,
              label: "Risk",
              number: savedPortfolioData.risk,
              format: "%",
              toggleLabel: "Max drawdown"
            }}
            secondValue={{
              tooltipContent: ToolTipContent.maxDrawdown,
              label: "Max drawdown",
              toggleLabel: "Risk",
              number: savedPortfolioData.maxDrawdown,
              format: "%"
            }}
          />
        </Row>
      </PortfolioSection>
    );
  return null;
}
