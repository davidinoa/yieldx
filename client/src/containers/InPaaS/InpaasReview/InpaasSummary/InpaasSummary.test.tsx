import React from "react";
import { render, screen } from "../../../../test/utils";
import InpaasSummary from "./InpaasSummary";

describe("InpaasSummary", () => {
  test("Should render correct info for inpaas review", () => {
    const { asFragment } = render(
      <InpaasSummary
        reviewData={{
          risk: 5,
          actualIncome: 550,
          actualYield: .25,
          actualAllocation: {
            maximumWeight: .25,
            minimumWeight: .03
          },
          maxDrawdown: 30
        }}
      />
    );
    expect(screen.queryAllByTestId("Portfolio-Metric-Card").length).toBe(4);
    expect(asFragment()).toMatchSnapshot();
  });

  test("Should render correct info for saved inpaas portfolio", () => {
    const { asFragment } = render(
      <InpaasSummary
        savedPortfolioData={{
          targetIncome: 550,
          targetYield: .25,
          actualYield: .30,
          actualIncome: 600,
          risk: 5,
          maxDrawdown: 30,
          expenseRatio: 5
        }}
      />
    );
    expect(screen.queryAllByTestId("Portfolio-Metric-Card").length).toBe(4);
    expect(asFragment()).toMatchSnapshot();
  });
});
