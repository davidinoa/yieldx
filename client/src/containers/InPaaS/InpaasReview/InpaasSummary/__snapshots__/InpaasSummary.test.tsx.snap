// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InpaasSummary Should render correct info for inpaas review 1`] = `
<DocumentFragment>
  .c5 > svg {
  margin: -0.5rem 0 0 0.35rem;
  vertical-align: middle;
  width: 0.85rem;
  height: 0.85rem;
}

.c0 {
  margin-bottom: 3rem;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

.c0 .analytics-card-container {
  min-height: 20rem;
  margin-bottom: 0.85rem;
}

.c0 .analytics-card-container > div {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
}

.c1 {
  padding-top: 0.5rem;
  padding-bottom: 0.5rem;
}

.c1 h2 {
  margin: 0;
  font-size: 1.15rem;
}

.c1 svg {
  margin: -0.5rem 0 0 0.35rem;
  vertical-align: middle;
  width: 0.85rem;
  height: 0.85rem;
}

.c2 span {
  border: solid 1px var(--border-new);
  color: var(--primary);
  border-radius: 0.25rem;
  font-weight: bold;
  padding: 0.25rem 0.75rem;
  font-size: 0.75rem;
}

.c2 span b {
  font-weight: bold;
  color: var(--lightest);
  -webkit-letter-spacing: 0.2em;
  -moz-letter-spacing: 0.2em;
  -ms-letter-spacing: 0.2em;
  letter-spacing: 0.2em;
  font-size: 0.5rem;
}

.c3 {
  width: 100%;
  height: 100%;
  background: var(--dark);
  border: 1px solid var(--border-new);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: stretch;
  -webkit-justify-content: stretch;
  -ms-flex-pack: stretch;
  justify-content: stretch;
  border-radius: 0.5rem;
  padding: 1rem 1.15rem;
  min-height: 6rem;
}

.c3 > * {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.c4 {
  font-weight: bold;
  line-height: normal;
  -webkit-letter-spacing: 0.2em;
  -moz-letter-spacing: 0.2em;
  -ms-letter-spacing: 0.2em;
  letter-spacing: 0.2em;
  text-transform: uppercase;
  color: var(--lightest);
  position: relative;
  font-size: 0.65rem;
  padding: 0.25rem 0;
  margin-bottom: 0.25rem;
  max-height: 2rem;
}

.c4 > svg {
  margin: -0.5rem 0 0 0.35rem;
  vertical-align: middle;
  width: 0.85rem;
  height: 0.85rem;
}

.c6 {
  line-height: normal;
  font-size: 1.5rem;
}

.c6 sup {
  font-size: 1rem;
  padding: 0.25rem;
}

.c6 span {
  color: var(--subtext-3);
  font-size: 12px;
  line-height: 15px;
}

.c6 .allocation-separator {
  height: 23px;
  width: 2px;
  background: var(--border-new);
}

.c7 {
  margin-left: auto;
  color: var(--lightest);
  font-size: 0.8rem;
  -webkit-text-decoration: underline;
  text-decoration: underline;
  padding: 0;
  background: none;
  border: none;
}

@media (min-width:1600px) {
  .c5 > svg {
    width: 1rem;
    height: 1rem;
  }
}

@media (min-width:1600px) {
  .c1 h2 {
    font-size: 1.5rem;
  }
}

@media (min-width:1600px) {
  .c1 svg {
    width: 1rem;
    height: 1rem;
  }
}

@media (min-width:1600px) {
  .c2 span {
    font-size: 0.9rem;
  }
}

@media (min-width:1600px) {
  .c2 span b {
    font-size: 0.725rem;
  }
}

@media (min-width:1600px) {
  .c3 {
    border-radius: 0.625rem;
    padding: 1rem 1.65rem 1.65rem 1.65rem;
  }
}

@media (min-width:1600px) {
  .c4 {
    font-size: 0.75rem;
    margin-bottom: 0.5rem;
  }
}

@media (min-width:1600px) {
  .c4 > svg {
    width: 1rem;
    height: 1rem;
  }
}

@media (min-width:1600px) {
  .c6 {
    font-size: 1.95rem;
  }
}

@media (min-width:1600px) {
  .c6 sup {
    font-size: 1.15rem;
  }
}

<div
    class="container-fluid"
  >
    <section
      class="c0 container-fluid"
    >
      <div
        class="mb-2 row"
      >
        <div
          class="c1 col-16"
        >
          <h2>
            SummaryÂ 
          </h2>
        </div>
        <div
          class="c2 col-8 d-flex align-items-center justify-content-end"
        />
      </div>
      <div
        class="row"
      >
        <div
          class="mb-1 col-6"
          data-testid="Portfolio-Metric-Card"
        >
          <div
            class="c3"
          >
            <div
              class="c4 d-flex align-items-center"
            >
              Yield
              <i
                class="c5"
              >
                <svg>
                  info.svg
                </svg>
              </i>
            </div>
            <div
              class="c6"
            >
              0.25
              <sup>
                %
              </sup>
              <span>
                /annual
              </span>
            </div>
          </div>
        </div>
        <div
          class="mb-1 col-6"
          data-testid="Portfolio-Metric-Card"
        >
          <div
            class="c3"
          >
            <div
              class="c4 d-flex align-items-center"
            >
              Income
              <i
                class="c5"
              >
                <svg>
                  info.svg
                </svg>
              </i>
              <button
                class="c7"
                type="button"
              >
                Annual
              </button>
            </div>
            <div
              class="c6"
            >
              $550
              <sup>
                .00
              </sup>
              <span>
                /monthly
              </span>
            </div>
          </div>
        </div>
        <div
          class="mb-1 col-6"
          data-testid="Portfolio-Metric-Card"
        >
          <div
            class="c3"
          >
            <div
              class="c4 d-flex align-items-center"
            >
              Risk
              <i
                class="c5"
              >
                <svg>
                  info.svg
                </svg>
              </i>
              <button
                class="c7"
                type="button"
              >
                Max Drawdown
              </button>
            </div>
            <div
              class="c6"
            >
              5.00
              <sup>
                %
              </sup>
            </div>
          </div>
        </div>
        <div
          class="mb-1 col-6"
          data-testid="Portfolio-Metric-Card"
        >
          <div
            class="c3"
          >
            <div
              class="c4 d-flex align-items-center"
            >
              Allocation
            </div>
            <div
              class="c6"
            >
              <div
                class="d-flex justify-content-between align-items-center"
              >
                <div>
                  3.00
                  <sup>
                    %
                  </sup>
                  <span>
                    /min
                  </span>
                </div>
                <div
                  class="allocation-separator"
                />
                <div>
                  25.00
                  <sup>
                    %
                  </sup>
                  <span>
                    /max
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </section>
  </div>
</DocumentFragment>
`;

exports[`InpaasSummary Should render correct info for saved inpaas portfolio 1`] = `
<DocumentFragment>
  .c5 > svg {
  margin: -0.5rem 0 0 0.35rem;
  vertical-align: middle;
  width: 0.85rem;
  height: 0.85rem;
}

.c0 {
  margin-bottom: 3rem;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

.c0 .analytics-card-container {
  min-height: 20rem;
  margin-bottom: 0.85rem;
}

.c0 .analytics-card-container > div {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
}

.c1 {
  padding-top: 0.5rem;
  padding-bottom: 0.5rem;
}

.c1 h2 {
  margin: 0;
  font-size: 1.15rem;
}

.c1 svg {
  margin: -0.5rem 0 0 0.35rem;
  vertical-align: middle;
  width: 0.85rem;
  height: 0.85rem;
}

.c2 span {
  border: solid 1px var(--border-new);
  color: var(--primary);
  border-radius: 0.25rem;
  font-weight: bold;
  padding: 0.25rem 0.75rem;
  font-size: 0.75rem;
}

.c2 span b {
  font-weight: bold;
  color: var(--lightest);
  -webkit-letter-spacing: 0.2em;
  -moz-letter-spacing: 0.2em;
  -ms-letter-spacing: 0.2em;
  letter-spacing: 0.2em;
  font-size: 0.5rem;
}

.c3 {
  width: 100%;
  height: 100%;
  background: var(--dark);
  border: 1px solid var(--border-new);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: stretch;
  -webkit-justify-content: stretch;
  -ms-flex-pack: stretch;
  justify-content: stretch;
  border-radius: 0.5rem;
  padding: 1rem 1.15rem;
  min-height: 6rem;
}

.c3 > * {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.c4 {
  font-weight: bold;
  line-height: normal;
  -webkit-letter-spacing: 0.2em;
  -moz-letter-spacing: 0.2em;
  -ms-letter-spacing: 0.2em;
  letter-spacing: 0.2em;
  text-transform: uppercase;
  color: var(--lightest);
  position: relative;
  font-size: 0.65rem;
  padding: 0.25rem 0;
  margin-bottom: 0.25rem;
  max-height: 2rem;
}

.c4 > svg {
  margin: -0.5rem 0 0 0.35rem;
  vertical-align: middle;
  width: 0.85rem;
  height: 0.85rem;
}

.c6 {
  line-height: normal;
  font-size: 1.5rem;
}

.c6 sup {
  font-size: 1rem;
  padding: 0.25rem;
}

.c6 span {
  color: var(--subtext-3);
  font-size: 12px;
  line-height: 15px;
}

.c6 .allocation-separator {
  height: 23px;
  width: 2px;
  background: var(--border-new);
}

.c7 {
  margin-left: auto;
  color: var(--lightest);
  font-size: 0.8rem;
  -webkit-text-decoration: underline;
  text-decoration: underline;
  padding: 0;
  background: none;
  border: none;
}

@media (min-width:1600px) {
  .c5 > svg {
    width: 1rem;
    height: 1rem;
  }
}

@media (min-width:1600px) {
  .c1 h2 {
    font-size: 1.5rem;
  }
}

@media (min-width:1600px) {
  .c1 svg {
    width: 1rem;
    height: 1rem;
  }
}

@media (min-width:1600px) {
  .c2 span {
    font-size: 0.9rem;
  }
}

@media (min-width:1600px) {
  .c2 span b {
    font-size: 0.725rem;
  }
}

@media (min-width:1600px) {
  .c3 {
    border-radius: 0.625rem;
    padding: 1rem 1.65rem 1.65rem 1.65rem;
  }
}

@media (min-width:1600px) {
  .c4 {
    font-size: 0.75rem;
    margin-bottom: 0.5rem;
  }
}

@media (min-width:1600px) {
  .c4 > svg {
    width: 1rem;
    height: 1rem;
  }
}

@media (min-width:1600px) {
  .c6 {
    font-size: 1.95rem;
  }
}

@media (min-width:1600px) {
  .c6 sup {
    font-size: 1.15rem;
  }
}

<div
    class="container-fluid"
  >
    <section
      class="c0 container-fluid"
    >
      <div
        class="mb-2 row"
      >
        <div
          class="c1 col-16"
        >
          <h2>
            SummaryÂ 
          </h2>
        </div>
        <div
          class="c2 col-8 d-flex align-items-center justify-content-end"
        />
      </div>
      <div
        class="row"
      >
        <div
          class="mb-1 col-6"
          data-testid="Portfolio-Metric-Card"
        >
          <div
            class="c3"
          >
            <div
              class="c4 d-flex align-items-center"
            >
              Current Yield
              <i
                class="c5"
              >
                <svg>
                  info.svg
                </svg>
              </i>
            </div>
            <div
              class="c6"
            >
              0.30
              <sup>
                %
              </sup>
              <span>
                /annual
              </span>
            </div>
          </div>
        </div>
        <div
          class="mb-1 col-6"
          data-testid="Portfolio-Metric-Card"
        >
          <div
            class="c3"
          >
            <div
              class="c4 d-flex align-items-center"
            >
              Current Income
              <i
                class="c5"
              >
                <svg>
                  info.svg
                </svg>
              </i>
            </div>
            <div
              class="c6"
            >
              $600
              <sup>
                .00
              </sup>
              <span>
                /monthly
              </span>
            </div>
          </div>
        </div>
        <div
          class="mb-1 col-6"
          data-testid="Portfolio-Metric-Card"
        >
          <div
            class="c3"
          >
            <div
              class="c4 d-flex align-items-center"
            >
              Expense ratio
              <i
                class="c5"
              >
                <svg>
                  info.svg
                </svg>
              </i>
            </div>
            <div
              class="c6"
            >
              5.00
              <sup>
                %
              </sup>
            </div>
          </div>
        </div>
        <div
          class="mb-1 col-6"
          data-testid="Portfolio-Metric-Card"
        >
          <div
            class="c3"
          >
            <div
              class="c4 d-flex align-items-center"
            >
              Risk
              <i
                class="c5"
              >
                <svg>
                  info.svg
                </svg>
              </i>
              <button
                class="c7"
                type="button"
              >
                Max Drawdown
              </button>
            </div>
            <div
              class="c6"
            >
              5.00
              <sup>
                %
              </sup>
            </div>
          </div>
        </div>
      </div>
    </section>
  </div>
</DocumentFragment>
`;
